### ISSUE TITLE
wxmaxima 17.10.0 fails to compile

### ISSUE THREAD

**Issue by badshah400** [andrejv/wxmaxima#929]
[_log.zip](https://github.com/andrejv/wxmaxima/files/1346646/_log.zip)

wxmaxima 17.10.0 fails to compile for all openSUSE versions, see [here](https://build.opensuse.org/package/show/home:badshah400:branches:Education/wxMaxima). A full build log when compiling for openSUSE Tumbleweed (`gcc7-7.2.1+r251579`) is attached. The typical error is of the form:

```
[  105s] /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/MathCtrl.cpp:205:22: error: no match for 'operator!=' (operand types are 'wxString' and 'wxToolTip*')
[  105s]            if(toolTip != GetToolTip())
[  105s]               ~~~~~~~~^~~~~~~~~~~~~~~
[  105s] In file included from /usr/include/wx-3.0/wx/memory.h:15:0,
[  105s]                  from /usr/include/wx-3.0/wx/object.h:19,
[  105s]                  from /usr/include/wx-3.0/wx/wx.h:15,
[  105s]                  from /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/wxMaximaFrame.h:32,
[  105s]                  from /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/wxMaxima.h:33,
[  105s]                  from /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/MathCtrl.cpp:31:
[  105s] /usr/include/wx-3.0/wx/string.h:3991:13: note: candidate: bool operator!=(const wxString&, const wxString&) <near match>
[  105s]  inline bool operator!=(const wxString& s1, const wxString& s2)
[  105s]              ^~~~~~~~
[  105s] /usr/include/wx-3.0/wx/string.h:3991:13: note:   conversion of argument 2 would be ill-formed:
[  105s] /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/MathCtrl.cpp:205:36: error: conversion from 'wxToolTip*' to 'const wxString' is ambiguous
[  105s]            if(toolTip != GetToolTip())
[  105s]                                     ^
[  105s] In file included from /usr/include/wx-3.0/wx/memory.h:15:0,
[  105s]                  from /usr/include/wx-3.0/wx/object.h:19,
[  105s]                  from /usr/include/wx-3.0/wx/wx.h:15,
[  105s]                  from /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/wxMaximaFrame.h:32,
[  105s]                  from /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/wxMaxima.h:33,
[  105s]                  from /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/MathCtrl.cpp:31:
[  105s] /usr/include/wx-3.0/wx/string.h:1235:3: note: candidate: wxString::wxString(wchar_t, size_t) <near match>
[  105s]    wxString(wchar_t ch, size_t nRepeat = 1)
[  105s]    ^~~~~~~~
[  105s] /usr/include/wx-3.0/wx/string.h:1235:3: note:   conversion of argument 1 would be ill-formed:
[  105s] /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/MathCtrl.cpp:205:35: error: invalid conversion from 'wxToolTip*' to 'wchar_t' [-fpermissive]
[  105s]            if(toolTip != GetToolTip())
[  105s]                          ~~~~~~~~~~^~
[  105s] In file included from /usr/include/wx-3.0/wx/memory.h:15:0,
[  105s]                  from /usr/include/wx-3.0/wx/object.h:19,
[  105s]                  from /usr/include/wx-3.0/wx/wx.h:15,
[  105s]                  from /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/wxMaximaFrame.h:32,
[  105s]                  from /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/wxMaxima.h:33,
[  105s]                  from /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/MathCtrl.cpp:31:
[  105s] /usr/include/wx-3.0/wx/string.h:1231:3: note: candidate: wxString::wxString(char, size_t) <near match>
[  105s]    wxString(char ch, size_t nRepeat = 1)
[  105s]    ^~~~~~~~
[  105s] /usr/include/wx-3.0/wx/string.h:1231:3: note:   conversion of argument 1 would be ill-formed:

```

when compiling with

```
cd /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/build/src && /usr/bin/c++  -DPREFIX=\"/usr\" -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -DwxDEBUG_LEVEL=0 -isystem /usr/lib64/wx/include/gtk2-unicode-3.0 -isystem /usr/include/wx-3.0 -I/home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/build/src  -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -DNDEBUG -pthread -O2 -g -DNDEBUG   -o CMakeFiles/wxmaxima.dir/MatrCell.cpp.o -c /home/abuild/rpmbuild/BUILD/wxmaxima-Version-17.10.0/src/MatrCell.cpp
```

**Comment by badshah400** [comment]
Thanks for the quick fix.

**Comment by gunterkoenigsmann** [comment]
You are welcome!

Unfortunately in the meantime I have  found out that this fix itself had a bug (a potential crash) that is closed by Commit eeea6de81d7d2259e8730da3d45d5df1fad75397
